src/shutdown/detach-dm.c

```diff
@@ -99,23 +99,31 @@ static int delete_dm(DeviceMapper *m) {
         assert(major(m->devnum) != 0);
         assert(m->path);
 
-        fd = open("/dev/mapper/control", O_RDWR|O_CLOEXEC);
-        if (fd < 0)
-                return -errno;
-
-        r = fsync_path_at(AT_FDCWD, m->path);
-        if (r < 0)
-                log_debug_errno(r, "Failed to sync DM block device %s, ignoring: %m", m->path);
-
-        return RET_NERRNO(ioctl(fd, DM_DEV_REMOVE, &(struct dm_ioctl) {
+        struct dm_ioctl dminfo = {
                 .version = {
                         DM_VERSION_MAJOR,
                         DM_VERSION_MINOR,
                         DM_VERSION_PATCHLEVEL
                 },
                 .data_size = sizeof(struct dm_ioctl),
                 .dev = m->devnum,
-        }));
+        };
+
+        fd = open("/dev/mapper/control", O_RDWR|O_CLOEXEC);
+        if (fd < 0)
+                return -errno;
+
+        r = ioctl(fd, DM_DEV_STATUS, &dminfo);
+        if (r < 0)
+                log_debug_errno(r, "Failed to get DM block device %s status, ignoring: %m", m->path);
+
+        if (dminfo.target_count) {
```

Comment:
Please add comment about why we need to check this in the code.

Reply:
in our codebase, when it comes to integers that are used in a numeric way (i.e. not as booleans or flags) we prefer if explicit numeric comparisons are made, instead of relying on C's downgrade-to-bool feature. or in other words, write this as `if (dminfo.target_count != 0) {…`